cmd/kubectl_token.go

```diff
@@ -381,18 +379,26 @@ func basicAuth(input *LoginInput) (managementClient.Token, error) {
 		responseType = fmt.Sprintf("%s_%s", responseType, input.clusterID)
 	}
 
-	body := fmt.Sprintf(`{"responseType":%q, "username":%q, "password":%q}`, responseType, username, password)
+	reqBody := fmt.Sprintf(`{"responseType":%q, "username":%q, "password":%q}`, responseType, username, password)
 
-	url := fmt.Sprintf("%s/v3-public/%ss/%s?action=login", input.server, input.authProvider,
+	loginURL := fmt.Sprintf("%s/v3-public/%ss/%s?action=login", input.server, input.authProvider,
 		strings.ToLower(strings.Replace(input.authProvider, "Provider", "", 1)))
 
-	response, err := request(http.MethodPost, url, bytes.NewBufferString(body))
+	req, err := http.NewRequest(http.MethodPost, loginURL, bytes.NewBufferString(reqBody))
+	if err != nil {
+		return token, fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, respBody, err := doRequest(client, req)
+	if err == nil && resp.StatusCode != http.StatusOK {
+		err = fmt.Errorf("unexpected http status code %d", resp.StatusCode)
+	}
 	if err != nil {
 		return token, nil
```

Comment:
I do not understand this here. We have an error here (see 2 lines above), and return the token, but not the error.
I would have expected
```suggestion
		return nil, err
```
here.

Reply:
Good catch, this is a left-over from the previous error  handling logic.
Changed in 3338ec579d191906c6765c1ab0eada15bb21d7a5. We now attempt to parse the the error response only if the request comes back with the status is not OK. The apiError unmarshalling error is ignored deliberately.